plugin/external/pluginEntry.go

```diff
@@ -40,42 +40,65 @@ type decodedExternalPluginEntry struct {
 
 const entryMethodTypeError = "each method must be a string or tuple [<method>, <result_or_signature>], not %v"
 
-func mungeToMethods(input []interface{}) (map[string]methodInfo, error) {
+type methodTuple struct {
+	Method            string
+	ResultOrSignature json.RawMessage
```

Comment:
This should be consistent with what we'll be documenting in the external plugin docs. I'm thinking `Value`'s more generic. Then I explain things as "value of the method tuple depends on the method. For list, it is list's result (prefetching). For read, it is read's signature OR its result. For exec, it is the transport ..."

Nit: May be worth changing `Method` => `MethodName` since external plugin docs we say `(<method_name>, ...)` when talking about the method tuple.

Reply:
`methodInfo` should probably change `result` => `tupleValue` so it is more generic.

Reply:
I said ResultOrSignature because that's explicitly what was used on the error string when you don't match the expected form.

Reply:
Yeah, that error string should probably be changed then.